bigtable/google/cloud/bigtable/table.py

```diff
@@ -671,45 +707,83 @@ class _RetryableMutateRowsWorker(object):
     )
     # pylint: enable=unsubscriptable-object
 
-    def __init__(self, client, table_name, rows, app_profile_id=None, timeout=None):
+    def __init__(
+        self, client, table_name, rows=[], batches=[], app_profile_id=None, timeout=None
+    ):
         self.client = client
         self.table_name = table_name
         self.rows = rows
+        if rows:
+            batches = [rows]
+        self.batches = dict([(index, batch) for index, batch in enumerate(batches)])
```

Comment:
What's the purpose of making a 2D list out of `rows`, i.e. if `rows = [some_values]` then `batches = [[some_values]]`? With this logic, `self.batches` will contain only a a single element, e.g.:

```
>>> a = [1,2,3,4,5]
>>> x = [a]
>>> batches = dict([(index, batch) for index, batch in enumerate(x)])
>>> print(batches)
{0: [1, 2, 3, 4, 5]}
```